link.go

```diff
@@ -227,7 +235,11 @@ func attachLink(s *Session, r *Receiver, opts []LinkOption) (*link, error) {
 		l.Messages = make(chan Message, l.receiver.maxCredit)
 		l.unsettledMessages = map[string]struct{}{}
 		// copy the received filter values
-		l.Source.Filter = resp.Source.Filter
+		if l.Source == nil {
```

Comment:
Completeness or was this just an outright bug?

Reply:
It's a bug.  A handful of link options initialize `l.Source` and presumably we always use one or more of these in "real" code.  The receiver test code doesn't use any of these options which is how I hit this.